src/Kunstmaan/NodeBundle/Helper/Services/PageCreatorService.php

```diff
@@ -95,118 +148,214 @@ public function setContainer(ContainerInterface $container = null)
+            $this->handleTranslation($translation);
         }
 
-        $em = $this->entityManager;
+        $this->aclPermissionCreatorService->createPermission($this->rootNode);
+        return $this->rootNode;
+    }
 
+    /**
+     * @param array $translations
+     */
+    private function throwExceptionIfNoTranslations(array $translations)
+    {
+        if (is_null($translations) || count($translations) == 0) {
+            throw new InvalidArgumentException('There has to be at least 1 translation in the translations array');
+        }
+    }
+
+    /**
+     * @return NodeRepository
+     */
+    private function getNodeRepo()
+    {
         /** @var NodeRepository $nodeRepo */
-        $nodeRepo = $em->getRepository('KunstmaanNodeBundle:Node');
-        /** @var $userRepo UserRepository */
-        $userRepo = $em->getRepository($this->userEntityClass);
+        $nodeRepo = $this->entityManager->getRepository('KunstmaanNodeBundle:Node');
+        return $nodeRepo;
+    }
+
+    /**
+     * @return UserRepository
+     */
+    private function getUserRepo()
+    {
+        /** @var UserRepository $userRepo */
+        $userRepo = $this->entityManager->getRepository($this->userEntityClass);
+        return $userRepo;
+    }
+
+    /**
+     * @return \Kunstmaan\NodeBundle\Repository\NodeTranslationRepository
+     */
+    private function getTranslationsRepo()
+    {
+        /* @var \Kunstmaan\NodeBundle\Repository\NodeTranslationRepository $nodeTranslationRepo*/
+        $nodeTranslationRepo = $this->entityManager->getRepository('KunstmaanNodeBundle:NodeTranslation');
+        return $nodeTranslationRepo;
+    }
+
+    /**
+     * @return \Kunstmaan\SeoBundle\Entity\Seo|null
+     */
+    private function getSeo()
+    {
+        $seo = null;
         /** @var $seoRepo SeoRepository */
         try {
-            $seoRepo = $em->getRepository('KunstmaanSeoBundle:Seo');
+            $seoRepo = $this->entityManager->getRepository('KunstmaanSeoBundle:Seo');
+            if (!is_null($seoRepo)) {
+                $seo = $seoRepo->findOrCreateFor($this->pageTypeInstance);
+            }
         } catch (ORMException $e) {
             $seoRepo = null;
         }
+        return $seo;
+    }
 
-        $pagecreator = array_key_exists('creator', $options) ? $options['creator'] : 'pagecreator';
-        $creator     = $userRepo->findOneBy(array('username' => $pagecreator));
+    /**
+     * @return User
+     */
+    private function getCreator()
+    {
+        $pagecreator = array_key_exists('creator', $this->options) ? $this->options['creator'] : 'pagecreator';
 
-        $parent = isset($options['parent']) ? $options['parent'] : null;
+        $userRepo = $this->getUserRepo();
 
-        $pageInternalName = isset($options['page_internal_name']) ? $options['page_internal_name'] : null;
+        /** @var User $creator */
+        $creator = $userRepo->findOneBy(array('username' => $pagecreator));
```

Comment:
```array()``` => ```[]```

Reply:
Argh do we need 5.3 array notation? I'll change it back! And sure, I'll move stuff into the constructor too